File: algorithms/[724]寻找数组的中心索引/solution.py
from typing import List


class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        # 总和
        total = sum(nums)
        # 前缀和
        pre_sum = 0

        for idx, ele in enumerate(nums):
            if (pre_sum * 2 + ele == total):
                return idx
            pre_sum += ele

        return -1

File: algorithms/[724]寻找数组的中心索引/solution.ts
// topics = ["数组"]

function pivotIndex(nums: number[]): number {
  const total = nums.reduce((a, b) => a + b, 0);
  let sum = 0;

  for (let [idx, ele] of nums.entries()) {
    if (sum * 2 + ele === total) {
      return idx;
    }
    sum += ele;
  }

  return -1;
}
